### PR TITLE
Reduce Test Time

### PR DESCRIPTION
**Pull request by adam-a-a**
## Summary/Motivation:
This PR eliminates/reduces tests in 0DRO and 1DRO that seemed like overkill and were significantly affecting time to complete testing. 

Additionally, a minor bug fix was applied to the 1DRO model's reporting method.

## Changes proposed in this PR:
- eliminate initialization arg tests in 0D and 1DRO models (handled by tests in util folder)
- reduce number of nodes to 3 for all 1dro tests (100 nodes used in one series of tests, and 20 nodes used in other tests)
- fix bug when specifying 3 nodes -->change the way the second index of `length_domain` is referenced

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: proteuslib/unit_models/tests/test_reverse_osmosis_1D.py [hunk 1]
```diff
@@ -461,27 +386,27 @@ def test_conservation(self, RO_frame):
     @pytest.mark.component
     def test_solution(self, RO_frame):
         m = RO_frame
-        x_interface_in = value(1 / m.fs.unit.nfe)
-        assert (pytest.approx(-1.5596e5, rel=1e-3) == value(m.fs.unit.deltaP_stage[0]))
-        assert (pytest.approx(1.219e-2, rel=1e-3) ==
+        x_interface_in = m.fs.unit.feed_side.length_domain[2]
+        assert (pytest.approx(-1.755e5, rel=1e-3) == value(m.fs.unit.deltaP_stage[0]))
+        assert (pytest.approx(7.992e-3, rel=1e-3) ==
```
Comment (TimBartholomew) [inline proteuslib/unit_models/tests/test_reverse_osmosis_1D.py:388]: Why did the inlet flux change? Shouldn't that basically only be based on the inlet conditions and independent on the number of nodes?

File: proteuslib/unit_models/tests/test_reverse_osmosis_1D.py [hunk 2]
```diff
@@ -461,27 +386,27 @@ def test_conservation(self, RO_frame):
...
                 value(m.fs.unit.flux_mass_phase_comp[0, x_interface_in, 'Liq', 'H2O']))
-        assert (pytest.approx(1.733e-6, rel=1e-3) ==
+        assert (pytest.approx(2.114e-6, rel=1e-3) ==
```
Comment (TimBartholomew) [inline proteuslib/unit_models/tests/test_reverse_osmosis_1D.py:390]: Same here

File: proteuslib/unit_models/tests/test_reverse_osmosis_1D.py [hunk 3]
```diff
@@ -620,20 +545,38 @@ def testReverseOsmosis_basic(self):
...
+              value(m.fs.unit.permeate_out[0].flow_mass_phase_comp['Liq', 'NaCl'])
+        )
+              # value(m.fs.unit.flux_mass_phase_comp_avg[0, 'Liq', 'H2O'] * 3.6e3),
+              # value(m.fs.unit.flux_mass_phase_comp_avg[0, 'Liq', 'NaCl'] * 3.6e6),
+              # value(m.fs.unit.permeate_out[0].flow_mass_phase_comp['Liq', 'H2O']),
+              # value(m.fs.unit.permeate_out[0].flow_mass_phase_comp['Liq', 'NaCl']),
+              # value(m.fs.unit.N_Re_avg[0]),
+              # value(m.fs.unit.Kf_avg[0, 'NaCl'] * 3.6e6),
+              # value(m.fs.unit.area)
+              # )
```
Comment (TimBartholomew) [inline proteuslib/unit_models/tests/test_reverse_osmosis_1D.py]: Commented code

**Review by andrewlee94 — APPROVED** [review]
Looks OK to me, as long as we have some large scale validation tests somewhere.

File: proteuslib/unit_models/tests/test_reverse_osmosis_1D.py [hunk 4]
```diff
@@ -770,20 +713,29 @@ def testReverseOsmosis_cp_mod_fixed(self):
...
+        x_interface_in = m.fs.unit.feed_side.length_domain[2]
+        print(
+              value(m.fs.unit.flux_mass_phase_comp[0, x_interface_in, 'Liq', 'H2O']),
+              value(m.fs.unit.flux_mass_phase_comp[0, x_interface_in, 'Liq', 'NaCl']),
+              value(m.fs.unit.area),
+              value(m.fs.unit.flux_mass_phase_comp[0, 1, 'Liq', 'H2O']),
+              value(m.fs.unit.flux_mass_phase_comp[0, 1, 'Liq', 'NaCl']),
+              value(m.fs.unit.permeate_out[0].flow_mass_phase_comp['Liq', 'H2O']),
+              value(m.fs.unit.permeate_out[0].flow_mass_phase_comp['Liq', 'NaCl'])
+        )
```
Comment (TimBartholomew) [inline proteuslib/unit_models/tests/test_reverse_osmosis_1D.py]: This print is for debugging right?

File: proteuslib/unit_models/tests/test_reverse_osmosis_1D.py [hunk 5]
```diff
@@ -916,20 +868,29 @@ def testReverseOsmosis_cp_calculated_kf_fixed(self):
...
```
Comment (TimBartholomew) [inline proteuslib/unit_models/tests/test_reverse_osmosis_1D.py]: This print too

File: proteuslib/unit_models/tests/test_reverse_osmosis_1D.py [hunk 6]
```diff
@@ -1077,20 +1038,29 @@ def testReverseOsmosis_cp_calculated_kf_calculated(self):
                 <= 1e-2)
 
         # Test solution
-        x_interface_in = value(1 / m.fs.unit.nfe)
-        assert (pytest.approx(4.419e-3, rel=1e-3) ==
+        x_interface_in = m.fs.unit.feed_side.length_domain[2]
+        print(
```
Comment (TimBartholomew) [inline proteuslib/unit_models/tests/test_reverse_osmosis_1D.py]: print

File: proteuslib/unit_models/tests/test_reverse_osmosis_1D.py [hunk 7]
```diff
@@ -1242,21 +1212,30 @@ def testRO_cp_calculated_kf_calculated_pdrop_fixed_by_dx(self):
         assert (abs(value(flow_mass_inlet - flow_mass_retentate - flow_mass_permeate)) <= 1e-2)
 
         # Test solution
-        x_interface_in = value(1 / m.fs.unit.nfe)
+        x_interface_in = m.fs.unit.feed_side.length_domain[2]
+        print(value(m.fs.unit.deltaP_stage[0]),
```
Comment (TimBartholomew) [inline proteuslib/unit_models/tests/test_reverse_osmosis_1D.py]: print

File: proteuslib/unit_models/tests/test_reverse_osmosis_1D.py [hunk 8]
```diff
@@ -1409,20 +1388,29 @@ def testRO_cp_calculated_kf_calculated_pdrop_fixed_by_stage(self):
                 <= 1e-2)
 
         # Test solution
-        x_interface_in = value(1 / m.fs.unit.nfe)
+        x_interface_in = m.fs.unit.feed_side.length_domain[2]
+        print(value(m.fs.unit.deltaP_stage[0]),
```
Comment (TimBartholomew) [inline proteuslib/unit_models/tests/test_reverse_osmosis_1D.py]: print

**Review by TimBartholomew — COMMENTED** [review]
Mostly comments on clean up. I'm confused why the inlet flux is changing though, I thought that would be independent of the number of nodes.

File: proteuslib/unit_models/tests/test_reverse_osmosis_1D.py (hunk 1 above)
Comment (adam-a-a) [reply proteuslib/unit_models/tests/test_reverse_osmosis_1D.py:388]: The flux at the "inlet" is not directly at the inlet. It is the second element in the continuous set (length_domain). So we should expect this value to change as it is not directly fixed by the inlet conditions. As a reminder, the actual inlet is skipped for most constraints, including those for membrane flux.

File: proteuslib/unit_models/tests/test_reverse_osmosis_1D.py (hunk 2 above)
Comment (adam-a-a) [reply proteuslib/unit_models/tests/test_reverse_osmosis_1D.py:390]: The same applies to this and anywhere else that uses `x_interface_in` as the space index--the index of the actual inlet is skipped.

**Comment by adam-a-a** [comment]
> Looks OK to me, as long as we have some large scale validation tests somewhere.

Well, I considered the test that I just reduced (from 100 nodes to 3 nodes) as a validation test. Are you suggesting we keep that test and throw it somewhere else?

**Comment by andrewlee94** [comment]
@adam-a-a Is that test comparing the results to some published data?

**Comment by adam-a-a** [comment]
> @adam-a-a Is that test comparing the results to some published data?

@andrewlee94 yes, it was comparing to one of @TimBartholomew 's papers

**Comment by TimBartholomew** [comment]
> > @adam-a-a Is that test comparing the results to some published data?
> 
> @andrewlee94 yes, it was comparing to one of @TimBartholomew 's papers

but now since we are using significantly less nodes it doesn't exactly match the results

**Comment by andrewlee94** [comment]
Exactly - that is why we have the different levels of tests. We should have some small, fast tests to make sure things build right, followed by some more thorough (but longer) tests that check that the results match literature data.

**Comment by TimBartholomew** [comment]
@andrewlee94 Oh I forgot one other thing, we don't exactly match the literature because we corrected the osmotic pressure calculation from being based on molarity to being based on molality. But we confirmed that the values are as expected with the correction. So its probably ok to just reduce all the tests to 3 nodes.

**Review by TimBartholomew — APPROVED** [review]

**Review by andrewlee94 — APPROVED** [review]

**Pull request merged**